### ISSUE TITLE
SwiftFormat does semantically not equivalent rewrite from `(#file)` to `#file`

### ISSUE THREAD

**Issue by weissi** [nicklockwood/SwiftFormat#638]
In Swift 5.3, `(#file)` can be used to silence a warning if you're passing `#file` to an argument that's defaulted to `#filePath`. `#filePath` however is Swift 5.3+ only so you can't easily use it and the shimming hacks [don't work](https://bugs.swift.org/browse/SR-12934).

Therefore, doing this

```
func foo(_ file: String = #filePath) {}

func bar(_ file: String = #file) {
   foo(file)
}
```

will in Swift 5.3 get you a warning. To silence it (because in Swift 5.3 `#file` == `#filePath`) you need to use:

```
func foo(_ file: String = #filePath) {}

func bar(_ file: String = (#file)) {
   foo(file)
}
```

SwiftFormat however removes the 'extra' set of parens.

**Comment by nicklockwood** [comment]
Thanks for reporting. I'll add a special case for this. In the meantime, adding this to the line above should fix it:

```swift
// swiftformat:disable:next redundantParens
```

**Comment by weissi** [comment]
https://github.com/apple/swift-log/pull/133

**Comment by nicklockwood** [comment]
@weissi fixed in 0.44.14.

**Comment by nicklockwood** [comment]
@weissi in lights of the changes in https://github.com/apple/swift-evolution/blob/master/proposals/0285-ease-pound-file-transition.md, is this exception still needed?
